compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConstValNotTopLevelOrObjectChecker.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.descriptors.ClassKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.declarations.FirRegularClass
+import org.jetbrains.kotlin.lexer.KtTokens
+
+object FirConstValNotTopLevelOrObjectChecker : FirPropertyChecker() {
+    override fun check(declaration: FirProperty, context: CheckerContext, reporter: DiagnosticReporter) {
+        val modifierList = with(FirModifierList) { declaration.source.getModifierList() }
+        if (modifierList?.modifiers?.any { it.token == KtTokens.CONST_KEYWORD } != true)
```

Comment:
Probably it's enough to check just `declaration.isConst` without struggling with `FirModifierList` (I will try this way myself).

Reply:
Yes, it works, and it's even better because CONST_VAL_NOT_TOP_LEVEL_OR_OBJECT should not be reported on local variables.